app/services/repository_datatable_service.rb

```diff
@@ -202,6 +204,36 @@ def build_name_filter_condition(repository_rows, filter_element_params)
     end
   end
 
+  def build_relationship_filter_condition(repository_rows, filter_element_params, index)
+    case filter_element_params[:operator]
+    when 'contains'
+      text = "%#{ActiveRecord::Base.sanitize_sql_like(filter_element_params.dig(:parameters, :text))}%"
+      repository_rows.joins("LEFT OUTER JOIN repository_row_connections repository_row_connections_#{index}
+                            ON repository_row_connections_#{index}.child_id = repository_rows.id
+                            LEFT OUTER JOIN repository_rows parent_repository_rows_#{index}
+                            ON parent_repository_rows_#{index}.id = repository_row_connections_#{index}.parent_id
+                            LEFT OUTER JOIN repository_row_connections child_connections_rows_#{index}
+                            ON child_connections_rows_#{index}.parent_id = repository_rows.id
+                            LEFT OUTER JOIN repository_rows child_repository_rows_#{index}
+                            ON child_repository_rows_#{index}.id = child_connections_rows_#{index}.child_id")
+                     .where("parent_repository_rows_#{index}.name ILIKE ? OR
+                            ('#{RepositoryRow::ID_PREFIX}' || parent_repository_rows_#{index}.id)::text ILIKE ? OR
+                            child_repository_rows_#{index}.name ILIKE ? OR
+                            ('#{RepositoryRow::ID_PREFIX}' || child_repository_rows_#{index}.id)::text ILIKE ?",
+                            text, text, text, text)
+    when 'contains_relationship'
+      repository_rows.left_outer_joins(:parent_connections, :child_connections)
```

Comment:
repository_rows = repository_rows.left_outer_joins(:parent_connections, :child_connections)
repository_rows.where.not(parent_connections: { id: nil }).or(repository_rows.where.not(child_connections: { id: nil }))